packages/location-field/src/index.tsx

```diff
@@ -645,57 +645,76 @@ const LocationField = ({
     const transitFeaturesPresent =
       stopFeatures.length > 0 || stationFeatures.length > 0;
 
-    const OtherFeaturesHeader = () => (
-      <S.MenuGroupHeader as={headingType} bgColor="#333" key="other-header">
-        <FormattedMessage
-          description="Text for header above the 'other' category of geocoder results"
-          id="otpUi.LocationField.other"
-        />
-      </S.MenuGroupHeader>
-    );
-
-    const otherFeaturesElements = otherFeatures.map(feature =>
-      renderFeature(itemIndex++, feature)
-    );
-
-    // Iterate through the geocoder results
-    menuItems = menuItems.concat(
-      renderOtherFirst &&
-        transitFeaturesPresent &&
-        otherFeatures.length > 0 && <OtherFeaturesHeader />,
-      renderOtherFirst && otherFeaturesElements,
-      stationFeatures.length > 0 && (
-        <S.MenuGroupHeader
-          as={headingType}
-          bgColor={layerColorMap.stations}
-          key="gtfs-stations-header"
-        >
+    const FeaturesElements = ({
+      bgColor,
+      key,
+      titleId,
+      featuresArray
+    }: {
+      bgColor: string;
+      key: string;
+      titleId: string;
+      featuresArray: JSX.Element[];
+    }) => {
+      const Header = () => (
+        <S.MenuGroupHeader as={headingType} bgColor={bgColor} key={key}>
           <FormattedMessage
-            description="Text for header above Stations"
-            id="otpUi.LocationField.stations"
+            description="Text for header above the 'other' category of geocoder results"
+            id={`otpUi.LocationField.${titleId}`}
           />
         </S.MenuGroupHeader>
-      ),
-      stationFeatures.map(feature => renderFeature(itemIndex++, feature)),
-
-      stopFeatures.length > 0 && (
-        <S.MenuGroupHeader
-          as={headingType}
-          bgColor={layerColorMap.stops}
-          key="gtfs-stops-header"
-        >
-          <FormattedMessage
-            description="Text for header above Stops"
-            id="otpUi.LocationField.stops"
+      );
+      return (
+        <>
+          {/* Only include the header if there are features to show */}
+          {titleId === "other" ? (
+            <Header />
+          ) : (
+            transitFeaturesPresent && <Header />
```

Comment:
We can probably shrink this down by combining these 2 conditions! `(titleId !== "other" || transitFeaturesPresent)`

Reply:
Nice!